### PR TITLE
[PORT] Flavor Roleplaying Overhaul

### PR DESCRIPTION
**Pull request by Bobbanz1**
## About The Pull Request
Expands on what was ported in this PR:
* https://github.com/BeeStation/NSV13/pull/2094

By porting the Following PRs from Skyrat-TG:
* https://github.com/Skyrat-SS13/Skyrat-tg/pull/675 (NSFW Warning - Only ported certain parts of this PR, the parts being those introduced in the current PR)
* https://github.com/Skyrat-SS13/Skyrat-tg/pull/1928 (Only the Silicon Flavor Text)
* https://github.com/Skyrat-SS13/Skyrat-tg/pull/6032
* https://github.com/Skyrat-SS13/Skyrat-tg/pull/7859 (Exploitable information not included)
* https://github.com/Skyrat-SS13/Skyrat-tg/pull/8350 (NSFW Warning - Don't completely remember everything but the one thing I know for sure I ported from this PR is the Examine Panel)
* https://github.com/Skyrat-SS13/Skyrat-tg/pull/8694 (Only some stuff related to the Examine Panel)
* https://github.com/Skyrat-SS13/Skyrat-tg/pull/8788
* https://github.com/Skyrat-SS13/Skyrat-tg/pull/15741
* https://github.com/Skyrat-SS13/Skyrat-tg/pull/15808
* https://github.com/Skyrat-SS13/Skyrat-tg/pull/16604 (Hybridized version of this PR as we do not have the map_view thingy)

Also random PR from TG which was needed for this thing:
* https://github.com/tgstation/tgstation/pull/58366 (Only the dummy generation thingy)

**ALSO**
Introduces a new character setup tab to contain all the relevant roleplaying stuff in it, what this means is that I have moved the Flavor Text to this new place, and placed everything else that was introduced in this PR (examine panel not included) in that preference tab.
<details>
<summary>Picture of New Tab</summary>

Screenshot 2023-01-13 200533

</details>

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

https://user-images.githubusercontent.com/59128051/212405149-959b88ab-69d3-4b55-8945-3034fc7e7d3d.mp4

</details>

## Changelog
:cl:Azarak, NotRanged, FlamingLily, nikothedude, Trigg, Iamgoofball, Gandalf2k15, GoldenAlpharex
add: Added flavor text functionality for silicons.
fix: General, Medical, and Security records work again.
add: Security HUDs and medical HUDs can now view sec/medical records on examine, respectively, and can also view general records.
add: Re-implemented silicon flavor texts!
code: Vastly improved the code of the Examine menu's Character Preview, making it a lot more reflective of what the character currently looks like, while also making it much more performant at the same time!
fix: Cyborgs will now appear in the Character Preview (without their lights, sadly) of the Examine panel too!
fix: The examine panel's character preview no longer shows you instead of the person you're looking at.
fix: The Examine Panel should now properly render characters, rather than just a select few mutant bodyparts.
fix: The Examine Panel's preview will now render characters always standing upright, and at 100% size, for consistency.
refactor: Flavor Examine thingy has been replaced with the Examine Panel and all code related to the previous one has been either removed outright or trimmed down.
/:cl:

### CODE REVIEW

**Review by Kmc2000 — COMMENTED** [review]
Since this involves DB queries, I am hesitant to touch this. Have you tested this on your local branch?

**Review by covertcorvid — DISMISSED** [review]

**Comment by Bobbanz1** [comment]
> This pull request has conflicts, please resolve those before we can evaluate the pull request.

Don't let the door hit you on the way out

**Review by Kmc2000 — COMMENTED** [review]
DB changes ahead..
@Bokkiewokkie any idea how we handle those? back in my day there was no db!

**Comment by covertcorvid** [comment]
> DB changes ahead.. @Bokkiewokkie any idea how we handle those? back in my day there was no db!

I know you have to load the new database schema before you run the testmerge, but I'm not sure how that's done without disrupting the current contents of the database

**Comment by Bobbanz1** [comment]
> This pull request has conflicts, please resolve those before we can evaluate the pull request.

Fuck it I'll fix you later

**Comment by Bokkiewokkie** [comment]
> DB changes ahead.. @Bokkiewokkie any idea how we handle those? back in my day there was no db!

Yeah it's very easy dw

File: tgui/packages/tgui/components/Box.tsx [hunk 1]
```diff
@@ -38,6 +38,7 @@ export interface BoxProps {
   bold?: BooleanLike;
   italic?: BooleanLike;
   nowrap?: BooleanLike;
+  preserveWhitespace?: BooleanLike;
```
Comment (Bokkiewokkie) [inline tgui/packages/tgui/components/Box.tsx:41]: Adding this here needs an NSV tag

File: tgui/packages/tgui/components/Box.tsx [hunk 2]
```diff
@@ -163,6 +164,7 @@ const styleMapperByPropName = {
   bold: mapBooleanPropTo('font-weight', 'bold'),
   italic: mapBooleanPropTo('font-style', 'italic'),
   nowrap: mapBooleanPropTo('white-space', 'nowrap'),
+  preserveWhitespace: mapBooleanPropTo('white-space', 'pre'),
```
Comment (Bokkiewokkie) [inline tgui/packages/tgui/components/Box.tsx:167]: Ditto for the tag

**Review by Bokkiewokkie — CHANGES_REQUESTED** [review]

File: code/datums/outfit.dm [hunk 3]
```diff
@@ -359,3 +359,36 @@
...
+		shoes = H.shoes.type
+	if(H.gloves)
+		gloves = H.gloves.type
+	if(H.ears)
+		ears = H.ears.type
+	if(H.glasses)
+		glasses = H.glasses.type
+	if(H.belt)
+		belt = H.belt.type
+	return TRUE
```
[the remark this answers is not in the document]
Comment (Bobbanz1) [reply code/datums/outfit.dm:394]: We don't apparently

**Review by Bokkiewokkie — APPROVED** [review]
🚀

**Pull request merged**